CTOR-2319-plugin-database-postgres-mode-locks-internal-postgre-sql-lock-types-virtualxid-transactionid-incorrectly-counted-as-exclusive-locks (#5572)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-databases-postgresql.md

Reviewer: ```suggestion
2. Whatever the license type (*online* or *offline*), install the **PostgreSQL DB** connector through
```

Reviewer: ```suggestion
The Monitoring Connector **PostgreSQL DB** brings a host template:
```

Reviewer 2: ```suggestion
The following monitoring connectors will be installed when you install the **PostgreSQL DB** connector through the
```

---
id: applications-databases-postgresql

> title: PostgreSQL

Reviewer: ```suggestion
title: PostgreSQL DB
description: "Monitor PostgreSQL databases: connections, cache hit ratio, locks, query time, vacuum, bloat, and tablespace size."
```

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias   | Service Template                       | Service Description                                                  | Discovery |
|:----------------|:---------------------------------------|:---------------------------------------------------------------------|:---------:|
| Bloat           | App-DB-Postgres-Bloat-custom           | Check tables and btrees bloat                                        |           |
| Database-Size   | App-DB-Postgres-Database-Size-custom   | Check the databases size                                             | X         |

> | Sql-Statement   | App-DB-Postgres-Sql-Statement-custom   | Check allowing to execute a custom SQL request with a digital answer |           |

Reviewer: ```suggestion
| Sql-Statement   | App-DB-Postgres-Sql-Statement-custom   | Check allowing to execute a custom SQL request with a numerical result |           |
```

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias   | Service Template                       | Service Description                                                  | Discovery |
|:----------------|:---------------------------------------|:---------------------------------------------------------------------|:---------:|
| Bloat           | App-DB-Postgres-Bloat-custom           | Check tables and btrees bloat                                        |           |
| Database-Size   | App-DB-Postgres-Database-Size-custom   | Check the databases size                                             | X         |
| Sql-Statement   | App-DB-Postgres-Sql-Statement-custom   | Check allowing to execute a custom SQL request with a numerical result |           |
| Statistics      | App-DB-Postgres-Statistics-custom      | Check database statistics                                            |           |

> | Tablespace-Size | App-DB-Postgres-Tablespace-Size-custom | Check time between poller and the Postgres server                    |           |

Reviewer 2: tablespace-size? doesn't match the description

> | Tablespace-Size | App-DB-Postgres-Tablespace-Size-custom | Check time between poller and the Postgres server                    |           |

Reviewer: ```suggestion
| Tablespace-Size | App-DB-Postgres-Tablespace-Size-custom | Check tablespace space usage                    |           |
```

> | Time-Sync       | App-DB-Postgres-Time-Sync-custom       | Check time between poller and the Postgres server                    |           |

Reviewer 2: ```suggestion
| Time-Sync       | App-DB-Postgres-Time-Sync-custom       | Check time offset between poller and Postgres server                    |           |
```

#### Service discovery

| Rule name                      | Description |
|:-------------------------------|:------------|

> | App-DB-Postgres-Databases-Size |             |

Reviewer: ```suggestion
| App-DB-Postgres-Databases-Size | Discover databases to monitor their sizes |
```

<Tabs groupId="sync">
<TabItem value="Bloat" label="Bloat">

| Name                                                | Unit |
|:----------------------------------------------------|:-----|
| *databases*~*tables*#table.space.usage.bytes        | B    |
| *databases*~*tables*#table.space.free.bytes         | B    |
| *databases*~*tables*#table.dead_tuple.bytes         | B    |
| *databases*~*indexes*#index.space.usage.bytes       | B    |
| *databases*~*indexes*#index.leaf_density.percentage | %    |

</TabItem>
<TabItem value="Cache-Hitratio" label="Cache-Hitratio">

| Name                                            | Unit |
|:------------------------------------------------|:-----|
| *hitratio*#database.hitratio.delta.percentage   | %    |
| *hitratio*#database.hitratio.average.percentage | %    |

</TabItem>
<TabItem value="Connection" label="Connection">

> | Metric name                  | Unit |

Reviewer: ```suggestion
| Name                | Unit |
```

### Using a host template provided by the connector

1. Log into Centreon and add a new host through **Configuration > Hosts**.
2. Fill in the **Name**, **Alias** & **IP Address/DNS** fields according to your resource's settings.
3. Apply the **App-DB-Postgres-custom** template to the host. A list of macros appears. Macros allow you to define how the connector will connect to the resource, and to customize the connector's behavior.
4. Fill in the macros you want. Some macros are mandatory.

| Macro                | Description                                                                                                                                        | Default value | Mandatory |
|:---------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|
| POSTGRESUSERNAME     |                                                                                                                                                    |               |           |

> | POSTGRESPASSWORD     |                                                                                                                                                    |               |           |

Reviewer: ```suggestion
| POSTGRESUSERNAME     | User name used to connect to the database                                                             |                   |      X      |
| POSTGRESPASSWORD     | Password for the defined user name                                                                    |                   |      X      |
```

| FILTERNAME    | Filter tablespace name after getting all tablespaces (can be a regexp) |               |           |
| TABLESPACE    | Filter tablespace name after getting all tablespaces (can be a regexp) |               |           |
| WARNING       | Threshold                                                              |               |           |
| CRITICAL      | Threshold                                                              |               |           |

</TabItem>
<TabItem value="Time-Sync" label="Time-Sync">

| Macro        | Description                                                                                                                                      | Default value | Mandatory |
|:-------------|:-------------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|
| WARNING      |                                                                                                                                                  | -1.0:1.0      |           |

> | CRITICAL     |                                                                                                                                                  | -3.0:3.0      |           |

Reviewer: ```suggestion
| WARNING      | Warning threshold in seconds. (use a range. it can be -0.3s or +0.3s.)                              | -1.0:1.0          |             |
| CRITICAL     | Critical threshold in seconds. (use a range. it can be -0.3s or +0.3s.)                             | -3.0:3.0          |             |
```

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).           |               |           |

</TabItem>
<TabItem value="Vacuum" label="Vacuum">

| Macro            | Description                                                                                                                                      | Default value | Mandatory |
|:-----------------|:-------------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|
| POSTGRESDATABASE | Database Name.                                                                                                                                   | postgres      |           |
| WARNING          |                                                                                                                                                  |               |           |

> | CRITICAL         |                                                                                                                                                  |               |           |

Reviewer: ```suggestion
| WARNING          | Warning threshold in seconds, maximum time interval since last vacuum                               |                   |             |
| CRITICAL         | Critical threshold in seconds, maximum time interval since last vacuum                              |                   |             |
```
